client/src/store/toast.ts

```diff
@@ -0,0 +1,56 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { IToastProps } from "../components/UI/Toast";
+
+
+export enum Title{
+    SUCCESS = "Success",
+    ERROR = "Error",
+}
+
+declare interface IPayload{
+    type: string;
+    message:string;
+}
+
+let toastProperties:IToastProps;
+
+const initialToastState: {toastProp: IToastProps[] | []} = {
+    toastProp: []
+}
+
+const toastSlice = createSlice({
+    name: 'toast',
+    initialState:  initialToastState,
+    reducers:{
+        showToast(state,action: PayloadAction<IPayload>){
+            switch(action.payload.type) {
+                case 'Success':
+                  toastProperties = {
+                    id: state.toastProp.length+1,
+                    title: Title.SUCCESS,
+                    description: action.payload.message,
+                    backgroundColor: '#5cb85c'
+                  }
+                  break;
+                case 'Error':
+                  toastProperties = {
+                    id: state.toastProp.length+1,
+                    title: Title.ERROR,
+                    description: action.payload.message,
+                    backgroundColor: '#d9534f'
+                  }
+                  break;
+                default:
+                  break ;
+              }
+            
+            state.toastProp = [...state.toastProp,toastProperties];
+        },
+        removeToast(state,action: PayloadAction<IToastProps[]>){
+            state.toastProp = action.payload;
```

Comment:
does this remove a single toast if so,. we should get the id as payload and remove that particular toast

Reply:
yes it removes a single toast.. Updated it, now it takes the id as payload.